h2o-hadoop/h2o-mapreduce-generic/src/main/java/water/hadoop/h2odriver.java

```diff
@@ -1301,6 +1301,13 @@ private int run2(String[] args) throws Exception {
     if (principal != null && keytabPath != null) {
       UserGroupInformation.setConfiguration(conf);
       UserGroupInformation.loginUserFromKeytab(principal, keytabPath);
+      // performs user impersonation (will only work if core-site.xml has hadoop.proxyuser.*.* props set on name node
+      if (runAsUser != null) {
+        System.out.println("Attempting to securely impersonate user, " + runAsUser);
+        UserGroupInformation currentEffUser = UserGroupInformation.getLoginUser();
+        UserGroupInformation proxyUser = UserGroupInformation.createProxyUser(runAsUser, currentEffUser);
+        UserGroupInformation.setLoginUser(proxyUser);
```

Comment:
This is fine in my opinion. We are already using this API for non-secured clusters.

Cleaner solution would look like this:

UserGroupInformation ugi = 
                     UserGroupInformation.createProxyUser(user, UserGroupInformation.getLoginUser());
             ugi.doAs(new PrivilegedExceptionAction<Void>() {
               public Void run() throws Exception {
                 //Submit a job
                 JobClient jc = new JobClient(conf);
                 jc.submitJob(conf);
                 //OR access hdfs
                 FileSystem fs = FileSystem.get(conf);
                 fs.mkdir(someFilePath); 
               }
             }

However I don't think it is necessary.